http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.java

```diff
@@ -120,11 +138,196 @@ private static URI poolKey(SdkHttpRequest sdkRequest) {
     private SslContext sslContext(String scheme) {
         if (scheme.equalsIgnoreCase("https")) {
             SslContextBuilder builder = SslContextBuilder.forClient().sslProvider(defaultClientProvider());
-            if (configuration.trustAllCertificates()) {
+            if (configuration.get(TRUST_ALL_CERTIFICATES)) {
                 builder.trustManager(InsecureTrustManagerFactory.INSTANCE);
             }
             return invokeSafely(builder::build);
         }
         return null;
     }
+
+    /**
+     * Factory that allows more advanced configuration of the Netty NIO HTTP implementation. Use {@link #builder()} to
+     * configure and construct an immutable instance of the factory.
+     */
+    @Immutable
+    public static final class Builder implements SdkAsyncHttpClient.Builder {
+
+        private final AttributeMap.Builder standardOptions = AttributeMap.builder();
+
+        private EventLoopGroup eventLoopGroup;
+        private EventLoopGroupFactory eventLoopGroupFactory;
+
+        private Builder() {
+        }
+
+        /**
+         * Max allowed connections per endpoint allowed in the connection pool.
+         *
+         * @param maxConnectionsPerEndpoint New value for max connections per endpoint.
+         * @return This builder for method chaining.
+         */
+        public Builder maxConnectionsPerEndpoint(Integer maxConnectionsPerEndpoint) {
+            standardOptions.put(MAX_CONNECTIONS, maxConnectionsPerEndpoint);
+            return this;
+        }
+
+        public void setMaxConnectionsPerEndpoint(Integer maxConnectionsPerEndpoint) {
+            maxConnectionsPerEndpoint(maxConnectionsPerEndpoint);
+        }
+
+        /**
+         * The maximum number of pending acquires allowed. Once this exceeds, acquire tries will be failed.
+         *
+         * @param maxPendingAcquires Max number of pending acquires
+         * @return This builder for method chaining.
+         */
+        public Builder maxPendingConnectionAcquires(Integer maxPendingAcquires) {
+            standardOptions.put(MAX_PENDING_CONNECTION_ACQUIRES, maxPendingAcquires);
+            return this;
+        }
+
+        public void setMaxPendingConnectionAcquires(Integer maxPendingAcquires) {
```

Comment:
Are we not doing the interface to hide the POJO setters anymore?

Reply:
We did it for client builders so that we can hide certain options for AWS vs non-AWS with the same core implementations. I didn't see the same benefit here, so I didn't bother doing it.

Reply:
Oh, I see what you mean. I'll switch to an interface.